tests/audio/test_sdr.py

```diff
@@ -148,3 +149,18 @@ def test_on_real_audio():
         rtol=0.0001,
         atol=1e-4,
     )
+
+
+def test_issue_895():
+    current_file_dir = os.path.dirname(__file__)
+    data = np.load(os.path.join(current_file_dir, "examples/issue_895.npz"))
+    preds = torch.tensor(data["preds"])
+    target = torch.tensor(data["target"])
+    sdr_tm = signal_distortion_ratio(preds, target)
```

Comment:
if we add the warning we will also be sure that the new code is actually triggered in this test
```suggestion
    with pytest.warns(UserWarning, match="Detected `nan` or `inf` value in computed metric, retrying computation in double precision"):
        sdr_tm = signal_distortion_ratio(preds, target)
```